docs/porting/connectivity/ble.md

```diff
@@ -0,0 +1,37 @@
+# Bluetooth Low Energy (BLE)
+
+Mbed OS uses the Cordio stack for its BLE host. We recommend porting the Cordio platform<!--is that what they call it?--> to your board, or porting at the HCI level<!--what's the distinction between porting the platform and porting the HCI level? If one is ported to the board, where is the other ported to? The chip? What's the difference between porting to a whole board v porting to a chip?--> as described in the HCI documentation.
+<!--we need a decision from Andy about the Cordio docs before we start linking. Ours are 18 months old!-->
+
+## Replacing the Cordio stack
+
+You can replace the Cordio stack by porting a complete alternative stack. You must do this at the API level<!--as opposed to which level?--> by:
+
+1. Disabling the Cordio compilation.
+1. Implementing all the user interfaces (Gap, GattServer, GattClient, SecurityManager).
+1. Providing a BLEInstaceBase that `createBLEInstance()` returns.
+
+### Disabling Cordio
+
+All the source files for Cordio are part of a library and are currently required by the BLE feature. Remove this requirement so that none of the Cordio implementations are compiled, otherwise they will conflict with your implementation.
+
+### Implementing API classes and BLEInstanceBase
```

Comment:
```suggestion
### Implementing the user interfaces
```

Reply:
BLEInstanceBase is not a user interface, it's the implementation of BLE used by the BLE class which is the user interface. BLEInstanceBase is the internal interface that allows the BLE class use your private implementation of the BLE stack.

Reply:
It's a porting interface, not user by users.

Reply:
But it also has its own header, doesn't it? Maybe we should move the bit about BLEIstanceBase to the next section?

Reply:
Well, they have different function (one is the user facing API, the other is an internal porting interface) but the porting process of implementing them is the same. Originally I had this as separate sections but they were identical (save for names) so I combined them